### ISSUE TITLE
make caret / typing indicator smooth

### ISSUE THREAD

**Issue by KraXen72** [Automattic/simplenote-electron#2512]
current in simplenote:
![Rec_2020_12 08_14;12](https://user-images.githubusercontent.com/21956756/101488088-6be86200-395f-11eb-9587-7565117460f4.gif)

proposed: (this example is in word 2016):
![Rec_2020_12 08_14;13](https://user-images.githubusercontent.com/21956756/101488190-92a69880-395f-11eb-994f-a123b22ef46c.gif)

**Comment by codebykat** [comment]
Hi @KraXen72! Thanks for this report. This is a very subtle difference, and I don't think it's something we have control over.

Can you try it on this page and let me know if it's the same there?

https://microsoft.github.io/monaco-editor/playground.html

**Comment by KraXen72** [comment]
it's not working even on that page. i think it's achievable with css, atleast on monkeytype.com (a typing test) they got it to work, just go into the settings and turn smooth caret on. i'm pretty sure it can be done with some css.

**Comment by codebykat** [comment]
Okay, so it turns out there's [an option for this in Monaco](https://github.com/Microsoft/vscode/pull/51197), which means we could support it easily. However I tried it and I don't much like the "feel". It feels pretty weird and it is not in line with the standards used in every other text editor and the web.

If we did add support for this, we'd need to make it a configurable option, so there's some additional complexity there.

For now I'm adding it to our list of feature requests and closing this issue (we use GitHub issues only to track bugs and our immediate roadmap). It's something we will consider adding in the future if we get a lot of requests for it.

Thanks again for the feedback.

Mostly for my own future reference, here's the code to test it in the Monaco playground:

```
monaco.editor.create(document.getElementById("container"), {
	value: "function hello() {\n\talert('Hello world!');\n}",
	language: "javascript",
        cursorSmoothCaretAnimation: true,
});
```
